File: c/eliuds-eggs/eliuds_eggs.c
#include "eliuds_eggs.h"

int egg_count(int d)
{
    int i = 0;

    while(d > 0)
    {
        if(d % 2 == 1)
        {
            i++;
        }
        d = d/2;
    }
    return i;
}

File: zig/eliuds-eggs/eliuds_eggs.zig
pub fn eggCount(number: usize) usize {
    var num: usize = number;
    var sum: usize = 0;
    while (num > 0) {
        if (num % 2 != 0) sum = sum + 1;
        num = num / 2;
    }
    return sum;
}
